src/v2/Apps/Consign/Routes/SubmissionFlow/ArtworkDetails/Components/LocationAutocomplete.tsx

```diff
@@ -0,0 +1,177 @@
+      window.__googleMapsCallback = () => {
+        setReady(true)
+      }
+      return
+    }
+
+    setReady(true)
+  }, [])
+
+  useEffect(() => {
+    // @ts-ignore
+    if (typeof google === "undefined" || !ready) return
+    // @ts-ignore
+    autocompleteServiceRef.current = new google.maps.places.AutocompleteService()
+  }, [ready])
+
+  const fetchSuggestions = async (searchQuery: string) => {
+    const res = await autocompleteServiceRef.current.getPlacePredictions({
+      input: searchQuery,
+      types: ["(cities)"],
+    })
+
+    return res?.predictions
+  }
+
+  const updateSuggestions = async (value: string) => {
+    setSuggestions([])
+    if (value.trim().length > 2) {
+      try {
+        setIsLoading(true)
+        const suggestions = await fetchSuggestions(value)
+        setIsError(false)
+        setIsLoading(false)
+        if (suggestions) {
+          setSuggestions(
+            suggestions.map(option => ({
+              text: option.description!,
+              value: option.place_id!,
+            }))
+          )
+        }
+      } catch {
+        setIsLoading(false)
+        setIsError(true)
+        onError()
+      }
+    }
+  }
+
+  const handleSuggestionsFetchRequested = useMemo(
+    () => debounce(updateSuggestions, DEBOUNCE_DELAY),
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    []
+  )
+
+  const handleChange = ({ target: { value } }) => {
+    setFieldTouched("location", false)
+    setFieldValue("locationId", "")
+    setFieldValue("location", value)
+    handleSuggestionsFetchRequested(value)
+  }
+
+  const handleClick = ({ currentTarget: { value } }) => {
+    setFieldTouched("location", false)
+    handleSuggestionsFetchRequested(value)
+  }
+
+  const handleClear = () => {
+    setSuggestions([])
+    setFieldValue("locationId", "")
+    setFieldValue("location", "")
+  }
+
+  const handleSelect = ({ text, value }: AutocompleteInputOptionType) => {
+    setFieldValue("locationId", value)
+    setFieldValue("location", text)
+  }
+
+  const handleClose = () => {
+    setFieldTouched("location")
+  }
+
+  const renderOption = (option: AutocompleteInputOptionType) => (
+    <Flex alignItems="center" p={1} width="100%">
+      <Text ml={1} variant="md">
+        {option.text}
+      </Text>
+    </Flex>
+  )
+
+  return initialized ? (
+    <AutocompleteInput
+      title="Location"
+      placeholder="Enter City Where Artwork Is Located"
+      data-test-id="autocomplete-location"
+      spellCheck={false}
+      loading={isLoading}
+      defaultValue={values.location}
+      error={values.location.trim() && touched.location && errors.locationId}
+      onChange={handleChange}
+      onClick={handleClick}
+      onClear={handleClear}
+      options={suggestions || []}
+      onSelect={handleSelect}
+      onClose={handleClose}
+      renderOption={renderOption}
+    />
+  ) : (
+    <Input title="Location" placeholder="Enter City Where Artwork Is Located" />
+  )
+}
```

Comment:
Yes, I know. In the `Submission flow,` we decided that the `Location` field should work as an `Artist` field. And there are a lot of differences between the location field in the `Collector  profile` and in the `Submission flow`

1. How suggestions open. `LocationAutocompleteInput` is `always open to the bottom`. But in `Submission flow`, we want to `open suggestions to the top` because the location field is at the very bottom
2. Error handling. In `Submission flow` we have error handling logic for API requests. 
3. In `Submission flow` user must choose a location from the suggestions, if not - we show an error
4. In `Submission flow` query starts only after 3rd character entered
5. And it looks different.

Here I'll add more if remember...

I suppose it requires discussion so I'll tag @iskounen